### ISSUE TITLE
Setting kubernetes.ingressEndpoint.publishedService in a non-watched namespace causes Traefik to panic

### ISSUE THREAD

**Issue by hb3b** [containous/traefik#4314]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Simply configuring the `publishedService` option under `kubernetes.ingressEndpoint` causes Traefik to crash at startup. The `hostname` option works as advertised and does not cause crashing.

### What did you expect to see?

Normal startup.

### What did you see instead?

```
{"level":"info","msg":"Using TOML configuration file /config/traefik.toml","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Traefik version v1.7.6 built on 2018-12-14_06:43:37AM","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Preparing server https \u0026{Address::443 TLS:\u003cnil\u003e Redirect:\u003cnil\u003e Auth:\u003cnil\u003e WhitelistSourceRange:[] WhiteList:\u003cnil\u003e Compress:true ProxyProtocol:0xc0001df2c0 ForwardedHeaders:0xc000388be0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Enabling ProxyProtocol for trusted IPs [10.196.0.0/16]","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Preparing server traefik \u0026{Address::8080 TLS:\u003cnil\u003e Redirect:\u003cnil\u003e Auth:\u003cnil\u003e WhitelistSourceRange:[] WhiteList:\u003cnil\u003e Compress:false ProxyProtocol:\u003cnil\u003e ForwardedHeaders:0xc000388c00} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Preparing server http \u0026{Address::80 TLS:\u003cnil\u003e Redirect:0xc000088080 Auth:\u003cnil\u003e WhitelistSourceRange:[] WhiteList:\u003cnil\u003e Compress:true ProxyProtocol:0xc0001df500 ForwardedHeaders:0xc000388bc0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Enabling ProxyProtocol for trusted IPs [10.196.0.0/16]","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Starting provider configuration.ProviderAggregator {}","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Starting server on :443","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Starting server on :8080","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Starting server on :80","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Starting provider *kubernetes.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":[\"kube-system\",\"routing-rules\",\"devops\"],\"LabelSelector\":\"traffic-type=vpn\",\"IngressClass\":\"\",\"IngressEndpoint\":{\"IP\":\"\",\"Hostname\":\"\",\"PublishedService\":\"routing/router-vpn-traefik\"}}","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"ingress label selector is: \"traffic-type=vpn\"","time":"2018-12-18T21:11:43Z"}
{"level":"info","msg":"Creating in-cluster Provider client","time":"2018-12-18T21:11:43Z"}
{"level":"error","msg":"Error in Go routine: runtime error: invalid memory address or nil pointer dereference","time":"2018-12-18T21:11:44Z"}
{"level":"error","msg":"Stack: goroutine 37 [running]:\nruntime/debug.Stack(0x26a5fe6, 0x17, 0xc0002c1760)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xa7\ngithub.com/containous/traefik/safe.defaultRecoverGoroutine(0x2224b80, 0x45c7d90)\n\t/go/src/github.com/containous/traefik/safe/routine.go:148 +0x89\ngithub.com/containous/traefik/safe.OperationWithRecover.func1.1(0xc00086be28)\n\t/go/src/github.com/containous/traefik/safe/routine.go:156 +0x56\npanic(0x2224b80, 0x45c7d90)\n\t/usr/local/go/src/runtime/panic.go:513 +0x1b9\ngithub.com/containous/traefik/provider/kubernetes.(*clientImpl).GetService(0xc0001b1da0, 0xc0000d4f60, 0x7, 0xc0000d4f68, 0x12, 0x2, 0x2, 0x1, 0xc0001e9340)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/client.go:198 +0x68\ngithub.com/containous/traefik/provider/kubernetes.(*Provider).updateIngressStatus(0xc0005225a0, 0xc000191558, 0x29ccea0, 0xc0001b1da0, 0xc0001e93c8, 0xc0005abb20)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:442 +0x1d1\ngithub.com/containous/traefik/provider/kubernetes.(*Provider).loadIngresses(0xc0005225a0, 0x29ccea0, 0xc0001b1da0, 0x1, 0x1, 0xc0006dede0)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:414 +0x5c5\ngithub.com/containous/traefik/provider/kubernetes.(*Provider).Provide.func1.1(0x0, 0x0)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:145 +0x421\ngithub.com/containous/traefik/safe.OperationWithRecover.func1(0x0, 0x0)\n\t/go/src/github.com/containous/traefik/safe/routine.go:160 +0x62\ngithub.com/containous/traefik/vendor/github.com/cenk/backoff.RetryNotify(0xc000308510, 0x29a05c0, 0xc000308520, 0x275d400, 0xc00066cc40, 0xc0003c6b10)\n\t/go/src/github.com/containous/traefik/vendor/github.com/cenk/backoff/retry.go:37 +0xa2\ngithub.com/containous/traefik/provider/kubernetes.(*Provider).Provide.func1(0xc00008a380)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:165 +0x12d\ngithub.com/containous/traefik/safe.(*Pool).Go.func1()\n\t/go/src/github.com/containous/traefik/safe/routine.go:78 +0x37\ngithub.com/containous/traefik/safe.GoWithRecover.func1(0x275d5a8, 0xc0001b1e00)\n\t/go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d\ncreated by github.com/containous/traefik/safe.GoWithRecover\n\t/go/src/github.com/containous/traefik/safe/routine.go:136 +0x49\n","time":"2018-12-18T21:11:44Z"}
{"level":"error","msg":"Provider connection error: panic in operation: %!s(\u003cnil\u003e); retrying in 749.594306ms","time":"2018-12-18T21:11:44Z"}
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Traefik version v1.7.6 built on 2018-12-14_06:43:37AM
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
    # traefik.toml
    logLevel = "INFO"
    defaultEntryPoints = ["http","https"]
    [entryPoints]
      [entryPoints.http]
      address = ":80"
      compress = true
        [entryPoints.http.proxyProtocol]
        trustedIPs = ["10.196.0.0/16"]
        [entryPoints.http.redirect]
          regex = "^http://(.*)"
          replacement = "https://$1"
      [entryPoints.https]
      address = ":443"
      compress = true
        [entryPoints.https.proxyProtocol]
        trustedIPs = ["10.196.0.0/16"]
      [entryPoints.traefik]
      address = ":8080"
    [kubernetes]
    namespaces = ["kube-system", "routing-rules", "devops"]
    labelselector = "traffic-type=vpn"
      [kubernetes.ingressEndpoint]
      publishedService = "routing/router-vpn-traefik"
    [traefikLog]
      format = "json"
    [accessLog]
      format = "common"
    [accessLog.fields]
      defaultMode = "keep"
    [accessLog.fields.names]
    [accessLog.fields.headers]
      defaultMode = "keep"
    [accessLog.fields.headers.names]
    [api]
      entryPoint = "traefik"
      dashboard = true
      [api.statistics]
        recentErrors = 100
    [metrics]
      [metrics.prometheus]
        entryPoint = "traefik"
        buckets = [0.1,0.2,0.5,0.8,1.0,2.5,5.0,7.5,10.0]

```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
{"level":"info","msg":"Using TOML configuration file /config/traefik.toml","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Traefik version v1.7.6 built on 2018-12-14_06:43:37AM","time":"2018-12-18T21:34:22Z"}
{"level":"debug","msg":"Global configuration loaded {\"LifeCycle\":{\"RequestAcceptGraceTimeout\":0,\"GraceTimeOut\":10000000000},\"GraceTimeOut\":0,\"Debug\":true,\"CheckNewVersion\":true,\"SendAnonymousUsage\":false,\"AccessLogsFile\":\"\",\"AccessLog\":{\"format\":\"common\",\"fields\":{\"defaultMode\":\"keep\",\"headers\":{\"defaultMode\":\"keep\"}}},\"TraefikLogsFile\":\"\",\"TraefikLog\":{\"format\":\"json\"},\"Tracing\":null,\"LogLevel\":\"\",\"EntryPoints\":{\"http\":{\"Address\":\":80\",\"TLS\":null,\"Redirect\":{\"regex\":\"^http://(.*)\",\"replacement\":\"https://$1\"},\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":true,\"ProxyProtocol\":{\"Insecure\":false,\"TrustedIPs\":[\"10.196.0.0/16\"]},\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}},\"https\":{\"Address\":\":443\",\"TLS\":null,\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":true,\"ProxyProtocol\":{\"Insecure\":false,\"TrustedIPs\":[\"10.196.0.0/16\"]},\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}},\"traefik\":{\"Address\":\":8080\",\"TLS\":null,\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}}},\"Cluster\":null,\"Constraints\":[],\"ACME\":null,\"DefaultEntryPoints\":[\"http\",\"https\"],\"ProvidersThrottleDuration\":2000000000,\"MaxIdleConnsPerHost\":200,\"IdleTimeout\":0,\"InsecureSkipVerify\":false,\"RootCAs\":null,\"Retry\":null,\"HealthCheck\":{\"Interval\":30000000000},\"RespondingTimeouts\":null,\"ForwardingTimeouts\":null,\"AllowMinWeightZero\":false,\"KeepTrailingSlash\":false,\"Web\":null,\"Docker\":null,\"File\":null,\"Marathon\":null,\"Consul\":null,\"ConsulCatalog\":null,\"Etcd\":null,\"Zookeeper\":null,\"Boltdb\":null,\"Kubernetes\":{\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":[\"kube-system\",\"routing-rules\",\"devops\"],\"LabelSelector\":\"traffic-type=vpn\",\"IngressClass\":\"\",\"IngressEndpoint\":{\"IP\":\"\",\"Hostname\":\"\",\"PublishedService\":\"routing/router-vpn-traefik\"}},\"Mesos\":null,\"Eureka\":null,\"ECS\":null,\"Rancher\":null,\"DynamoDB\":null,\"ServiceFabric\":null,\"Rest\":null,\"API\":{\"EntryPoint\":\"traefik\",\"Dashboard\":true,\"Debug\":true,\"CurrentConfigurations\":null,\"Statistics\":{\"RecentErrors\":100}},\"Metrics\":{\"Prometheus\":{\"Buckets\":[0.1,0.2,0.5,0.8,1,2.5,5,7.5,10],\"EntryPoint\":\"traefik\"},\"Datadog\":null,\"StatsD\":null,\"InfluxDB\":null},\"Ping\":null,\"HostResolver\":null}","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n","time":"2018-12-18T21:34:22Z"}
{"level":"debug","msg":"Configured Prometheus metrics","time":"2018-12-18T21:34:22Z"}
{"level":"debug","msg":"Creating regex redirect http -\u003e ^http://(.*) -\u003e https://$1","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Preparing server http \u0026{Address::80 TLS:\u003cnil\u003e Redirect:0xc000089840 Auth:\u003cnil\u003e WhitelistSourceRange:[] WhiteList:\u003cnil\u003e Compress:true ProxyProtocol:0xc00039f860 ForwardedHeaders:0xc000598b20} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Enabling ProxyProtocol for trusted IPs [10.196.0.0/16]","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Preparing server https \u0026{Address::443 TLS:\u003cnil\u003e Redirect:\u003cnil\u003e Auth:\u003cnil\u003e WhitelistSourceRange:[] WhiteList:\u003cnil\u003e Compress:true ProxyProtocol:0xc00039fa20 ForwardedHeaders:0xc000598b40} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Enabling ProxyProtocol for trusted IPs [10.196.0.0/16]","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Preparing server traefik \u0026{Address::8080 TLS:\u003cnil\u003e Redirect:\u003cnil\u003e Auth:\u003cnil\u003e WhitelistSourceRange:[] WhiteList:\u003cnil\u003e Compress:false ProxyProtocol:\u003cnil\u003e ForwardedHeaders:0xc000598b00} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Starting provider configuration.ProviderAggregator {}","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Starting server on :80","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Starting server on :443","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Starting server on :8080","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Starting provider *kubernetes.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":[\"kube-system\",\"routing-rules\",\"devops\"],\"LabelSelector\":\"traffic-type=vpn\",\"IngressClass\":\"\",\"IngressEndpoint\":{\"IP\":\"\",\"Hostname\":\"\",\"PublishedService\":\"routing/router-vpn-traefik\"}}","time":"2018-12-18T21:34:22Z"}
{"level":"debug","msg":"Using Ingress label selector: \"traffic-type=vpn\"","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"ingress label selector is: \"traffic-type=vpn\"","time":"2018-12-18T21:34:22Z"}
{"level":"info","msg":"Creating in-cluster Provider client","time":"2018-12-18T21:34:22Z"}
{"level":"debug","msg":"Received Kubernetes event kind *v1beta1.Ingress","time":"2018-12-18T21:34:23Z"}
{"level":"error","msg":"Error in Go routine: runtime error: invalid memory address or nil pointer dereference","time":"2018-12-18T21:34:23Z"}
{"level":"error","msg":"Stack: goroutine 40 [running]:\nruntime/debug.Stack(0x26a5fe6, 0x17, 0xc000326490)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0xa7\ngithub.com/containous/traefik/safe.defaultRecoverGoroutine(0x2224b80, 0x45c7d90)\n\t/go/src/github.com/containous/traefik/safe/routine.go:148 +0x89\ngithub.com/containous/traefik/safe.OperationWithRecover.func1.1(0xc0008c9e28)\n\t/go/src/github.com/containous/traefik/safe/routine.go:156 +0x56\npanic(0x2224b80, 0x45c7d90)\n\t/usr/local/go/src/runtime/panic.go:513 +0x1b9\ngithub.com/containous/traefik/provider/kubernetes.(*clientImpl).GetService(0xc0002b91d0, 0xc0000df600, 0x7, 0xc0000df608, 0x12, 0x2, 0x2, 0x1, 0xc0004fa000)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/client.go:198 +0x68\ngithub.com/containous/traefik/provider/kubernetes.(*Provider).updateIngressStatus(0xc0007ce5a0, 0xc00018f2b0, 0x29ccea0, 0xc0002b91d0, 0xc0004fa088, 0xc00094d300)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:442 +0x1d1\ngithub.com/containous/traefik/provider/kubernetes.(*Provider).loadIngresses(0xc0007ce5a0, 0x29ccea0, 0xc0002b91d0, 0x1, 0x1, 0xc0006c3200)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:414 +0x5c5\ngithub.com/containous/traefik/provider/kubernetes.(*Provider).Provide.func1.1(0x0, 0x0)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:145 +0x421\ngithub.com/containous/traefik/safe.OperationWithRecover.func1(0x0, 0x0)\n\t/go/src/github.com/containous/traefik/safe/routine.go:160 +0x62\ngithub.com/containous/traefik/vendor/github.com/cenk/backoff.RetryNotify(0xc000593d90, 0x29a05c0, 0xc000593da0, 0x275d400, 0xc0005bebc0, 0xc000543f00)\n\t/go/src/github.com/containous/traefik/vendor/github.com/cenk/backoff/retry.go:37 +0xa2\ngithub.com/containous/traefik/provider/kubernetes.(*Provider).Provide.func1(0xc00064a310)\n\t/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:165 +0x12d\ngithub.com/containous/traefik/safe.(*Pool).Go.func1()\n\t/go/src/github.com/containous/traefik/safe/routine.go:78 +0x37\ngithub.com/containous/traefik/safe.GoWithRecover.func1(0x275d5a8, 0xc0002b9470)\n\t/go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d\ncreated by github.com/containous/traefik/safe.GoWithRecover\n\t/go/src/github.com/containous/traefik/safe/routine.go:136 +0x49\n","time":"2018-12-18T21:34:23Z"}
{"level":"error","msg":"Provider connection error: panic in operation: %!s(\u003cnil\u003e); retrying in 335.106995ms","time":"2018-12-18T21:34:23Z"}
```

**Comment by ldez** [comment]
The stacktrace inside the log is:
```go
Stack: goroutine 40 [running]:
runtime/debug.Stack(0x26a5fe6, 0x17, 0xc000326490)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
github.com/containous/traefik/safe.defaultRecoverGoroutine(0x2224b80, 0x45c7d90)
	/go/src/github.com/containous/traefik/safe/routine.go:148 +0x89
github.com/containous/traefik/safe.OperationWithRecover.func1.1(0xc0008c9e28)
	/go/src/github.com/containous/traefik/safe/routine.go:156 +0x56
panic(0x2224b80, 0x45c7d90)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/containous/traefik/provider/kubernetes.(*clientImpl).GetService(0xc0002b91d0, 0xc0000df600, 0x7, 0xc0000df608, 0x12, 0x2, 0x2, 0x1, 0xc0004fa000)
	/go/src/github.com/containous/traefik/provider/kubernetes/client.go:198 +0x68
github.com/containous/traefik/provider/kubernetes.(*Provider).updateIngressStatus(0xc0007ce5a0, 0xc00018f2b0, 0x29ccea0, 0xc0002b91d0, 0xc0004fa088, 0xc00094d300)
	/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:442 +0x1d1
github.com/containous/traefik/provider/kubernetes.(*Provider).loadIngresses(0xc0007ce5a0, 0x29ccea0, 0xc0002b91d0, 0x1, 0x1, 0xc0006c3200)
	/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:414 +0x5c5
github.com/containous/traefik/provider/kubernetes.(*Provider).Provide.func1.1(0x0, 0x0)
	/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:145 +0x421
github.com/containous/traefik/safe.OperationWithRecover.func1(0x0, 0x0)
	/go/src/github.com/containous/traefik/safe/routine.go:160 +0x62
github.com/containous/traefik/vendor/github.com/cenk/backoff.RetryNotify(0xc000593d90, 0x29a05c0, 0xc000593da0, 0x275d400, 0xc0005bebc0, 0xc000543f00)
	/go/src/github.com/containous/traefik/vendor/github.com/cenk/backoff/retry.go:37 +0xa2
github.com/containous/traefik/provider/kubernetes.(*Provider).Provide.func1(0xc00064a310)
	/go/src/github.com/containous/traefik/provider/kubernetes/kubernetes.go:165 +0x12d
github.com/containous/traefik/safe.(*Pool).Go.func1()
	/go/src/github.com/containous/traefik/safe/routine.go:78 +0x37
github.com/containous/traefik/safe.GoWithRecover.func1(0x275d5a8, 0xc0002b9470)
	/go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
created by github.com/containous/traefik/safe.GoWithRecover
	/go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
 ```

The root cause (L198):
https://github.com/containous/traefik/blob/7a963e75fdbd4edc96907b23e02b2d00fca844df/provider/kubernetes/client.go#L196-L201

**Comment by hb3b** [comment]
@ldez Not sure I understand. The service definitely does exist.

`k get svc router-vpn-traefik -n routing -o yaml`

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout: "60"
    service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
    service.beta.kubernetes.io/aws-load-balancer-extra-security-groups: sg-XXX
    service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: '*'
    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:us-east-1:XXX:certificate/XXX
    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "443"
  creationTimestamp: 2018-11-28T18:17:39Z
  labels:
    app: traefik
    chart: traefik-1.52.7
    heritage: Tiller
    release: router-vpn
  name: router-vpn-traefik
  namespace: routing
  resourceVersion: "7402846"
  selfLink: /api/v1/namespaces/routing/services/router-vpn-traefik
  uid: e40be8ce-f339-11e8-b61c-0effefd4f780
spec:
  clusterIP: 100.70.67.0
  externalTrafficPolicy: Local
  healthCheckNodePort: 30060
  loadBalancerSourceRanges:
  - XXX/22
  - XXX/22
  - XXX/32
  - XXX/32
  - XXX/32
  ports:
  - name: http
    nodePort: 30806
    port: 80
    protocol: TCP
    targetPort: http
  - name: https
    nodePort: 30668
    port: 443
    protocol: TCP
    targetPort: 443
  - name: metrics
    nodePort: 31524
    port: 8080
    protocol: TCP
    targetPort: dash
  selector:
    app: traefik
    release: router-vpn
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - hostname: XXX.us-east-1.elb.amazonaws.com
```

**Comment by ldez** [comment]
We found the issue: the namespace of the published service is not in the list of the namespaces to watch:
```yml
 namespaces = ["kube-system", "routing-rules", "devops"] # <---
    labelselector = "traffic-type=vpn"
      [kubernetes.ingressEndpoint]
      publishedService = "routing/router-vpn-traefik" # <---
 ```

**Comment by hb3b** [comment]
Thanks! That was a good find. Our team would benefit from being able to have the `publishedService` setting reference a service in an unwatched namespace. There are a number of ingress records in our `routing` namespace that shouldn't be available from this instance of Traefik.

**Comment by traefiker** [comment]
Closed by #4327.
